module/VuFind/tests/integration-tests/src/VuFindTest/Mink/SsoTest.php

```diff
@@ -0,0 +1,117 @@
+<?php
+
+/**
+ * Mink SSO test class.
+ *
+ * PHP version 8
+ *
+ * Copyright (C) Hebis Verbundzentrale 2023.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * @category VuFind
+ * @package  Tests
+ * @author   Thomas Wagener <wagener@hebis.uni-frankfurt.de>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org Main Page
+ */
+
+namespace VuFindTest\Mink;
+
+/**
+ * Mink SSO test class.
+ *
+ * @category VuFind
+ * @package  Tests
+ * @author   Thomas Wagener <wagener@hebis.uni-frankfurt.de>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org Main Page
+ * @retry    4
+ */
+class SsoTest extends \VuFindTest\Integration\MinkTestCase
+{
+    /**
+     * Get config.ini override settings for testing SSO login.
+     *
+     * @return array
+     */
+    public function getConfigIniOverrides()
+    {
+        return [
+            'config' => [
+                'Authentication' => [
+                    'method' => 'SimulatedSSO',
+                ],
+            ],
+        ];
+    }
+
+    /**
+     * Test changing a password.
+     *
+     * @retryCallback tearDownAfterClass
+     *
+     * @return void
+     */
+    public function testLogin()
+    {
+        // Set up configs
+        $this->changeConfigs($this->getConfigIniOverrides());
+        $session = $this->getMinkSession();
+        $session->visit($this->getVuFindUrl());
+        $page = $session->getPage();
+
+        // Login
+        $this->clickCss($page, '#loginOptions a');
+
+        // Log out
+        $this->clickCss($page, '.logoutOptions a.logout');
+    }
+
+    /**
+     * SSO login in lightbox
+     *
+     * @return void
+     */
+    public function testLightboxLogin()
+    {
+        // Set up configs
+        $this->changeConfigs($this->getConfigIniOverrides());
+        $session = $this->getMinkSession();
+        $session->visit($this->getVuFindUrl() . '/Record/testsample1');
+        $page = $session->getPage();
+
+        // Try to save record to list
+        $this->clickCss($page, '.record-nav .save-record');
+
+        // Login in lightbox
+        $this->clickCss($page, '.modal-body .btn.btn-link');
```

Comment:
Maybe assert something about  this link before clicking it, like confirm it says "log in" (or whatever it actually says).

Reply:
Done